app/uk/gov/hmrc/economiccrimelevyregistration/services/RegistrationValidationService.scala

```diff
@@ -16,79 +16,131 @@
+class RegistrationValidationService @Inject() (clock: Clock, schemaValidator: SchemaValidator) {
 
-  type ValidationResult[A] = ValidatedNec[DataValidationError, A]
+  type ValidationResult[A] = ValidatedNel[DataValidationError, A]
 
-  def validateRegistration(registration: Registration): ValidationResult[String] =
+  def validateRegistration(registration: Registration): ValidationResult[EclSubscription] =
+    transformToEclSubscription(registration) match {
+      case Valid(eclSubscription) =>
+        schemaValidator.validateAgainstJsonSchema(
+          eclSubscription,
+          SchemaLoader.loadSchema("create-ecl-subscription-request.json")
+        )
+      case invalid                => invalid
+    }
+
+  private def transformToEclSubscription(registration: Registration): ValidationResult[EclSubscription] =
     (
-      validateGrsJourneyData(registration),
-      validateAmlRegulatedActivity(registration),
+      validateLegalEntityDetails(registration),
       validateAmlSupervisor(registration),
-      validateOptExists(registration.relevantAp12Months, missingErrorMessage("Relevant AP 12 months choice")),
-      validateOptExists(registration.relevantApRevenue, missingErrorMessage("Relevant AP revenue")),
+      validateOptExists(registration.businessSector, "Business sector"),
+      validateFirstContactDetails(registration.contacts.firstContactDetails),
+      validateSecondContactDetails(registration),
+      validateEclAddress(registration.contactAddress),
+      validateAmlRegulatedActivity(registration),
+      validateOptExists(registration.relevantAp12Months, "Relevant AP 12 months choice"),
+      validateOptExists(registration.relevantApRevenue, "Relevant AP revenue"),
       validateConditionalOptExists(
         registration.relevantApLength,
         registration.relevantAp12Months.contains(false),
-        missingErrorMessage("Relevant AP length")
+        "Relevant AP length"
       ),
-      validateRevenueMeetsThreshold(registration),
-      validateOptExists(registration.contacts.firstContactDetails.name, missingErrorMessage("First contact name")),
-      validateOptExists(registration.contacts.firstContactDetails.role, missingErrorMessage("First contact role")),
+      validateRevenueMeetsThreshold(registration)
+    ).mapN {
+      (
+        legalEntityDetails,
+        amlSupervisor,
+        businessSector,
+        firstContactDetails,
+        secondContactDetails,
+        contactAddress,
+        _,
+        _,
+        _,
+        _,
+        _
+      ) =>
+        EclSubscription(
+          legalEntityDetails = legalEntityDetails(amlSupervisor, businessSector.toString),
+          correspondenceAddressDetails = contactAddress,
+          primaryContactDetails = firstContactDetails,
+          secondaryContactDetails = secondContactDetails
+        )
+    }
+
+  private def validateFirstContactDetails(details: ContactDetails): ValidationResult[SubscriptionContactDetails] =
```

Comment:
Could this be refactored and renamed to validateContactDetails so that it accepts a ContactDetails instance along with a flag to indicate if this instance is the first contact (true) or second contact (false) that could be used to construct the validation error text?

Reply:
I've pushed a commit which does this in a similar way, let me know if you're happy with that